test/libyul/yulOptimizerTests/reasoningBasedSimplifier/arith_movable.yul

```diff
@@ -0,0 +1,19 @@
+{
+	function f() -> z
+	{
+		z := 15
+	}
+	let x := 7
+	let y := 8
+	if eq(add(x, y), f()) { }
```

Comment:
So for this we'd need to inline non-recursive calls to movable functions?

Reply:
This step in isolation does not do function summaries, but the inliner will of course inline the function.

Reply:
It might still make sense to just descend into functions in this step (just treating recursive calls as fresh undefined values), since the inliner won't inline everything... but not sure and definitely not in this PR.